### PR TITLE
Make CI reject spelling errors using codespell + fix three typos

### PR DESCRIPTION
**Pull request by hartwork**
This is a port of https://github.com/libexpat/libexpat/pull/739 to ttyplot. Should help keep our typo count down in the future.

### CODE REVIEW

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2023 Sebastian Pipping <sebastian@pipping.org>
+# Licensed under Apache License Version 2.0
+
+name: Enforce codespell-clean spelling
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: '0 3 * * 5'  # Every Friday at 3am
```
Comment (edgar-bonet) [inline .github/workflows/codespell.yml:10]: Why the schedules run? Is that in case the word list changes?

**Review by edgar-bonet — APPROVED** [review]

File: .github/workflows/codespell.yml (hunk 1 above)
Comment (hartwork) [reply .github/workflows/codespell.yml:10]: @edgar-bonet the general idea that even if we try to pin things down as much as possible (which is the most secure and most robust approach meaning tas furthest away from flakey), some part of the CI ground below us is always moving and if we keep running CI once a week, we will never be more than two weeks behind noticing CI broke in the mean time (and hence get a chance to fix things in time and not when we want to release some hotfix a.s.a.p.), even if we do not do new pull requests for a few weeks more. I haven't checked in detail yet how much movement is inside the codespell action despite pinning. What do you think?

Comment (edgar-bonet) [reply .github/workflows/codespell.yml:10]: I see. makes sense.

Comment (hartwork) [reply .github/workflows/codespell.yml:10]: @edgar-bonet cool! :tada:

**Pull request merged**